### ISSUE TITLE
"command aws.amazonq.rejectCodeSuggestion not found" if connection expired

### ISSUE THREAD

**Issue by DustyLamp** [aws/aws-toolkit-vscode#3509]
## Problem
`command 'aws.codeWhisperer.rejectCodeSuggestion' not found` keeps popping up when I press escape when a suggestion is made and I want to dismiss it.

Note: I'm using Copilot and Code Whisperer at the same time.

## System details (run the `AWS: About Toolkit` command)
I'm using copilot and aws code whisperer at the same time.

-   OS: mac
-   Visual Studio Code version: 1.78.2
-   AWS Toolkit version: 1.74.0

**Comment by disbelief** [comment]
This happens for me even when I've explicitly disabled code whisperer in the AWS extension config

**Comment by dreamorosi** [comment]
Hi @justinmk3 I'm using AWS Toolkit 1.76 like you suggested in the previous comment but I'm still seeing the error, even though I have disabled code whisperer in the workspace:

<img width="463" alt="image" src="https://github.com/aws/aws-toolkit-vscode/assets/7353869/a3e8020c-15b8-45e9-9954-b8a754c85015">


<img width="506" alt="Screenshot 2023-06-10 at 12 17 12" src="https://github.com/aws/aws-toolkit-vscode/assets/7353869/a8e6a34a-588c-4e92-92be-4d5e5da215fe">

Only thing left to do is disable the extension entirely.

**Comment by Will-ShaoHua** [comment]
hi @dreamorosi, by "I have disable codewhisperer in the workspace", do you mean you have signed out from either Aws-builderId or Aws-Identity-center and also curious at that point of time, what's the UI you can see on CodeWhisperer node in the toolkit panel? Thanks!

**Comment by Will-ShaoHua** [comment]
hi @disbelief, could you elaborate more on the step you mentioned ` disabled code whisperer in the AWS extension config`, what config did you change then? Thanks!

**Comment by disbelief** [comment]
@Will-ShaoHua I actually ended up just disabling the entire AWS extension for my workspace. I thought it was possible to disable code whisperer in the extension settings, but you can only turn off some of its features:

<img width="622" alt="Screenshot 2023-06-11 at 09 03 56" src="https://github.com/aws/aws-toolkit-vscode/assets/295574/ba4b1eed-d753-4c51-b081-f492ef375066">

However after updating to the latest version of the AWS extension (and restarting VS Code) I'm not having this problem anymore.

**Comment by Will-ShaoHua** [comment]
Hi @disbelief, gotcha, yea I think we might need to make our UI more intuitive so that ppl won't be confused when they plan to disable it :) . So "disable" codewhisperer in the current product spec, we will be referring to "disconnect" from codewhisperer. Thus we should click on the smal unplugl icon next to the connection shown. 

![Screenshot 2023-06-11 at 2 29 10 AM](https://github.com/aws/aws-toolkit-vscode/assets/96078566/836354c8-e295-46f0-88c3-760a828d5432)


However this issue will still happen if there are multiple inline suggestion providers, thus if you enable both CodeWhisperer and Copilot at the same time, you will still see this error, which we are actively tracking. But you shouldn't see this error pop out when you are disconnected from codewhisperer. If that's happening then we have a bug lol

**Comment by dreamorosi** [comment]
Hey @Will-ShaoHua, in my case the connection expired and I didn't log in again. 

While it's true that I didn't actually remove the connection, I'm also not connected so I would expect CodeWhisperer to be disabled & not setting any event listener:

![Screenshot 2023-06-11 at 11 59 30](https://github.com/aws/aws-toolkit-vscode/assets/7353869/23125361-c728-4ff2-8588-2f49ea064754)

**Comment by Will-ShaoHua** [comment]
@dreamorosi understood, yea agree, that's a valid point, we should fix this. Thanks for the reply, that's really helpful!

**Comment by kisamoto** [comment]
I'm trying to use Code Whisperer but also receiving this meaning I cannot dismiss suggestions (very frustrating). 

AWS Toolkit v1.86.0
OS: Linux
VSCode version: 1.81.1

**Comment by Will-ShaoHua** [comment]
@kisamoto do you have other inline suggestions provider (e.g. Copilot, Gitlab  extensions etc.) installed? also could you provide me a screenshot of CodeWhisperer panel in your VSC editor? thanks

**Comment by leigaol** [comment]
@kisamoto can you try the latest AWS Toolkit for VS Code release? It should have fixed this issue for you.

**Comment by kisamoto** [comment]
@leigaol I'm afraid I have now switched from CodeWhisperer to Copilot and don't have the capacity to try to replicate. Sorry about that.

**Comment by rduan** [comment]
I can confirm this error is still there, I am using the AWS Toolkit  v2.8.0 with mac , vscode 1.86.0

**Comment by dgokcin** [comment]
I can still see the error on both vs code and vscode-insiders with the latest version of the plugin. How can I just disable the code whisperer suggestions but continue using the aws toolkit plugin?

**Comment by Will-ShaoHua** [comment]
hey @rduan, thanks for reporting to us. 
One thing I want to verify with you do you have other completion provider extension installed in your VSCode, e.g. Copilot, Tabnine etc. Asking because this error is reproducible by installing these kinds of extensions. Thanks!

**Comment by quisido** [comment]
>do you have other completion provider extension installed in your VSCode

I can confirm that this happens when Google Cloud Code Duet AI is installed.

**Comment by IgorGanapolsky** [comment]
This is still happening in 2024, in VS Code 1.89.1

**Comment by justinmk3** [comment]
@IgorGanapolsky using the `Amazon Q: About` command, can you share the exact version of Q you are using? also `AWS: About` if you are using AWS Toolkit.

**Comment by dgokcin** [comment]
For the ones coming across this: You can check my workaround in https://github.com/aws/aws-toolkit-vscode/issues/4409#issuecomment-1935540523 for a workaround.

**Comment by IgorGanapolsky** [comment]
@justinmk3 
> OS: Darwin arm64 23.5.0
Visual Studio Code extension host:  1.89.1
Amazon Q:  1.6.0
node: 18.18.2
electron: 28.2.8

**Comment by Will-ShaoHua** [comment]
@IgorGanapolsky  curious do you happen to have other inline suggestion extensions installed in your IDE? e.g. copilot, tabnine, gitlab workflow etc. We're aware of this bug can be reproed by conflicting inline suggestion providers

**Comment by IgorGanapolsky** [comment]
@Will-ShaoHua I definitely have all of the aforementioned code assistants installed.  But none of them give me the error that Q does......

**Comment by Will-ShaoHua** [comment]
@IgorGanapolsky, thanks for the confirmation. 

> none of them give me the error that Q does......
Yea on this part, i would say the root cause is our own implementation therefore you won't see same issue from other AI extensions...

**Comment by nagdeepanjan** [comment]
This issue is there there. I am getting the error: "command 'aws.amazonq.rejectCodeSuggestion' not found"
I am using VS Code 1.90.
I am using Amazon Q 1.9.0

I am using CoPilot.
I am using Amazon Q but PAUSED it and unchecked all options. I mostly use Amazon Q's chatbot feature.
I use Google IDX too and have the same problem.

**Comment by dgokcin** [comment]
you can try https://github.com/aws/aws-toolkit-vscode/issues/4409#issuecomment-1935540523 for a workaround

**Comment by dgokcin** [comment]
@nagdeepanjan, you can add the following to your `settings.json` to unmap the Esc key for Amazon Q as a workaround. However, I think this behavior needs to change. It does not make any sense for a disabled extension to error out like this.

```json
  {
    "key": "escape",
    "command": "-aws.amazonq.rejectCodeSuggestion",
    "when": "aws.codewhisperer.connected && inlineSuggestionVisible && !editorReadonly"
  }

```

**Comment by deepanjannag** [comment]
> @nagdeepanjan, you can add the following to your `settings.json` to unmap the Esc key for Amazon Q as a workaround. However, I think this behavior needs to change. It does not make any sense for a disabled extension to error out like this. If you are using vim/neovim, it makes the extension unusable.
> 
> ```json
>   {
>     "key": "escape",
>     "command": "-aws.amazonq.rejectCodeSuggestion",
>     "when": "aws.codewhisperer.connected && inlineSuggestionVisible && !editorReadonly"
>   }
> ```

Thank you so much!

**Comment by sascharo** [comment]
> @nagdeepanjan, you can add the following to your `settings.json` to unmap the Esc key for Amazon Q as a workaround. However, I think this behavior needs to change. It does not make any sense for a disabled extension to error out like this. If you are using vim/neovim, it makes the extension unusable.
> 
> ```json
>   {
>     "key": "escape",
>     "command": "-aws.amazonq.rejectCodeSuggestion",
>     "when": "aws.codewhisperer.connected && inlineSuggestionVisible && !editorReadonly"
>   }
> ```

> > @nagdeepanjan, you can add the following to your `settings.json` to unmap the Esc key for Amazon Q as a workaround. However, I think this behavior needs to change. It does not make any sense for a disabled extension to error out like this. If you are using vim/neovim, it makes the extension unusable.
> > ```json
> >   {
> >     "key": "escape",
> >     "command": "-aws.amazonq.rejectCodeSuggestion",
> >     "when": "aws.codewhisperer.connected && inlineSuggestionVisible && !editorReadonly"
> >   }
> > ```
> 
> Thank you so much!

Do you mean adding it to keybindings.json?

**Comment by sascharo** [comment]
> @Will-ShaoHua I definitely have all of the aforementioned code assistants installed. But none of them give me the error that Q does......

Agreed, Amazon Q is the only one.

**Comment by dgokcin** [comment]
@sascharo yes `keybindings.json` not `settings.json`. I edited my answer 🙃

**Comment by rmaclean** [comment]
Same issue

AWS toolkit v3.18.0
Amazon Q v1.19.0
No other inline tools/ai tools (like copilot) and no CodeWhisperer (what ever that is)
MacOS

<img width="366" alt="image" src="https://github.com/user-attachments/assets/ec191d88-36a4-4409-ace4-8e8704f257ff">

I can see it in the settings - pretty surer this is literally just a :
<img width="1111" alt="image" src="https://github.com/user-attachments/assets/ae363183-3237-4d4a-bdd1-d7b1977bc68b">

My workaround was to right click on that and select remove binding

**Comment by leigaol** [comment]
Should be fixed by https://github.com/aws/aws-toolkit-vscode/pull/5482. I tested a build after this fix with Copilot installed and I can no longer reproduce the command not found issue.

**Comment by leigaol** [comment]
It is fixed in release 1.23.0 Amazon Q for VS Code. Please let us know if you encounter this issue again. Thanks!
